### ISSUE TITLE
zero_pad in GatedGraphConv 

### ISSUE THREAD

**Issue by Rosarubu** [dmlc/dgl#2355]
Hi, I have a question about  GatedGraphConv layer, why you pad the input feature like this, that means the out_feature's size must  larger than the input feature?  
https://github.com/dmlc/dgl/blob/9c08cd6bf055eea253cd91d1e3765576004fd9a7/python/dgl/nn/pytorch/conv/gatedgraphconv.py#L144-L145

**Comment by yzh119** [comment]
Yes, the input tensor would be padded to the same shape as output so that we can use GRU to aggregate h^(i)

![image](https://user-images.githubusercontent.com/11773619/99938488-da93c000-2da2-11eb-980b-e571a2dfdf2c.png)

**Comment by Rosarubu** [comment]
In the paper, x_v is called node annotations, which seems more like a mask to me. Or I might be misunderstanding? 
Also, in another implementation https://github.com/calebmah/ggnn.pytorch/blob/master/model.py, zero padding is not found.

**Comment by yzh119** [comment]
I think they did the padding here: https://github.com/calebmah/ggnn.pytorch/blob/master/utils/train.py#L10.
The annotations should be viewed as node features in the bAbI task mentioned in the original paper.

**Comment by Rosarubu** [comment]
Thanks! Guess I need to go back to read the paper again :p

**Comment by yzh119** [comment]
Welcome to come back to us if you have further questions :)
